src/js/hmm/querys.ts

```diff
@@ -23,3 +25,15 @@ export function useListHmms(page: number, per_page: number, term?: string) {
         keepPreviousData: true,
     });
 }
+
+/**
+ * Fetches a single HMM
+ *
+ * @param hmmId - The id of the hmm to fetch
+ * @returns A single HMM
+ */
+export function useFetchHmm(hmmId: string) {
+    return useQuery<HMM>(hmmQueryKeys.detail(hmmId), () => fetchHmm(hmmId), {
+        retry: 0,
```

Comment:
I assume this is to handle a 404? I would pass a callback instead that looks for the correct error and bails.